router/batchrouter/asyncdestinationmanager/bing-ads/bulk_uploader.go

```diff
@@ -0,0 +1,296 @@
+		eventsLimit:   common.GetBatchRouterConfigInt64("MaxEventsLimit", destName, 4000000),
+	}
+}
+
+/*
+This function create at most 3 zip files from the text file created by the batchrouter
+It takes the text file path as input and returns the zip file path
+The maximum size of the zip file is 100MB, if the size of the zip file exceeds 100MB then the job is marked as failed
+*/
+func (b *BingAdsBulkUploader) Upload(asyncDestStruct *common.AsyncDestinationStruct) common.AsyncUploadOutput {
+	destination := asyncDestStruct.Destination
+	destConfig := DestinationConfig{}
+	jsonConfig, err := stdjson.Marshal(destination.Config)
+	if err != nil {
+		return common.AsyncUploadOutput{
+			FailedJobIDs:  append(asyncDestStruct.FailedJobIDs, asyncDestStruct.ImportingJobIDs...),
+			FailedReason:  fmt.Sprintf("got error while marshalling the destination config %v", err.Error()),
+			FailedCount:   len(asyncDestStruct.FailedJobIDs) + len(asyncDestStruct.ImportingJobIDs),
+			DestinationID: destination.ID,
+		}
+	}
+	err = stdjson.Unmarshal(jsonConfig, &destConfig)
+	if err != nil {
+		return common.AsyncUploadOutput{
+			FailedJobIDs:  append(asyncDestStruct.FailedJobIDs, asyncDestStruct.ImportingJobIDs...),
+			FailedReason:  fmt.Sprintf("got error while unmarshalling the destination config %v", err.Error()),
+			FailedCount:   len(asyncDestStruct.FailedJobIDs) + len(asyncDestStruct.ImportingJobIDs),
+			DestinationID: destination.ID,
+		}
+	}
+	var failedJobs []int64
+	var successJobs []int64
+	var importIds []string
+	var errors []string
+	actionFiles, err := b.CreateZipFile(asyncDestStruct.FileName, destConfig.AudienceID)
+	if err != nil {
+		return common.AsyncUploadOutput{
+			FailedJobIDs:  append(asyncDestStruct.FailedJobIDs, asyncDestStruct.ImportingJobIDs...),
+			FailedReason:  fmt.Sprintf("got error while transforming the file. %v", err.Error()),
+			FailedCount:   len(asyncDestStruct.FailedJobIDs) + len(asyncDestStruct.ImportingJobIDs),
+			DestinationID: destination.ID,
+		}
+	}
+	uploadRetryableStat := stats.Default.NewTaggedStat("events_over_prescribed_limit", stats.CountType, map[string]string{
+		"module":   "batch_router",
+		"destType": b.destName,
+	})
+	for _, actionFile := range actionFiles {
+		uploadRetryableStat.Count(len(actionFile.FailedJobIDs))
+		_, err := os.Stat(actionFile.ZipFilePath)
+		if err != nil || actionFile.EventCount == 0 {
+			continue
+		}
+		urlResp, err := b.service.GetBulkUploadUrl()
+		if err != nil {
+			b.logger.Error("Error in getting bulk upload url: %w", err)
+			failedJobs = append(append(failedJobs, actionFile.SuccessfulJobIDs...), actionFile.FailedJobIDs...)
+			errors = append(errors, fmt.Sprintf("%s:error in getting bulk upload url: %s", actionFile.Action, err.Error()))
+			continue
+		}
+
+		if urlResp.UploadUrl == "" || urlResp.RequestId == "" {
+			b.logger.Error(`{"error" : "getting empty string in upload url or request id. Provide correct configuration credentials."}`)
```

Comment:
Should this be checked in the application or do we let bing-ads-go-sdk handle these cases too post unmarshal

Reply:
@koladilip

Reply:
we should handle in bing-ads-go-sdk too. @ItsSudip can you have look into it?

Reply:
yes we can verify this along with error parsing why the SDK failed to get uploadUrl.

Reply:
addressed